File: apps/desktop/src-tauri/src/application/dashboard_charts_service.rs
use std::sync::Arc;

use crate::domain::entities::{DailyActivity, TopUser};
use crate::domain::ports::DashboardChartsRepository;

pub struct DashboardChartsService {
    repo: Arc<dyn DashboardChartsRepository>,
}

impl DashboardChartsService {
    pub fn new(repo: Arc<dyn DashboardChartsRepository>) -> Self {
        Self { repo }
    }

    pub async fn get_activity_trend(
        &self,
        guild_id: Option<String>,
        days: Option<i32>,
    ) -> Result<Vec<DailyActivity>, String> {
        self.repo.get_activity_trend(guild_id, days).await
    }

    pub async fn get_top_users(
        &self,
        guild_id: String,
        limit: Option<u32>,
    ) -> Result<Vec<TopUser>, String> {
        self.repo.get_top_users(guild_id, limit).await
    }
}

File: sentinel-web/src/services/dashboardChartsService.ts
import { httpGet } from "@/api/http";
import type { DailyActivity, TopUser } from "@/types";
import { q } from "./_query";

export const dashboardChartsService = {
  getActivityTrend(guildId?: string | null, days = 30): Promise<DailyActivity[]> {
    return httpGet(`/api/charts/activity${q({ guild_id: guildId ?? null, days })}`);
  },
  getTopUsers(guildId: string, limit = 10): Promise<TopUser[]> {
    return httpGet(`/api/stats/${guildId}/leaderboard${q({ limit })}`);
  },
};
